From d057cdaeb4bf5d1c091ec8fec76f6514fbf9a235 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Tue, 11 Feb 2020 17:07:28 +0200
Subject: [PATCH 080/269] s32g274a: Move bl31 at the top of the first 2GB

The previous location at the very top of physical memory (4GB, on the
S32G274A-EVB) is not reachable by the 32-bit eDMA, forcing us to fall
back to the slower memcpy.

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-4595
---
 plat/s32g/include/platform_def.h | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/plat/s32g/include/platform_def.h b/plat/s32g/include/platform_def.h
index 5cace495b..96a5cafa9 100644
--- a/plat/s32g/include/platform_def.h
+++ b/plat/s32g/include/platform_def.h
@@ -107,10 +107,12 @@
  */
 #define S32G_DDR0_END		0x8ffffffff
 
-/* Protected zone in DDR - we'll deploy BL31 there. This must also be kept
- * in sync with U-Boot, which is expected to alter the Linux device-tree.
+/* Protected zone in DDR - we'll deploy BL31 there. Choose the top of the first
+ * 2GB, which is reachable by the 32-bit eDMA.
+ * This must also be kept in sync with U-Boot, which is expected to alter
+ * the Linux device-tree.
  */
-#define S32G_PMEM_END		(S32G_DDR0_END)
+#define S32G_PMEM_END		(0xffffffff)
 #define S32G_PMEM_LEN		0x00200000	/* conservatively allow 2MB */
 #define S32G_PMEM_START		(S32G_PMEM_END - S32G_PMEM_LEN + 1)
 
-- 
2.17.1

